src/trunk.jl

```diff
@@ -116,7 +116,22 @@ function TrunkSolver(
   return TrunkSolver{T, V, Sub, Op}(x, xt, gx, gt, gn, Hs, subsolver, H, tr)
 end
 
-function LinearOperators.reset!(::TrunkSolver) end
+function SolverCore.reset!(solver::TrunkSolver{T, V, Sub, Op}) where {T, V, Sub, Op}
```

Comment:
Do you really want to generate a new `reset!` for each combination of parameters?